packages/common/container-definitions/src/fluidPackage.ts

```diff
@@ -86,6 +86,7 @@ export interface IFluidCodeDetails {
 
     /**
      * Configuration details. This includes links to the package manager and base CDNs.
+     * TODO: what does it mean for this to be undefined?
```

Comment:
Generally speaking, the code details are a user-defined pile of data that pairs with a user-defined code loader, to help that code loader find the appropriate code to load.  This `config` speculates that a common scenario might be to have a key/value pairing for things like CDNs or other tools the code loader might need to use to find the code.  However, since the code loader is user-defined we can't know whether that's the case or not -- so it might be undefined if the code loader has no need of these extra info.

I have issue #8176 with some commentary on this that might help add some detail.